kubeflow/trainer/rhai/progression_instrumentation.py

```
@@ -0,0 +1,267 @@
+
+
+class ProgressCallback(TrainerCallback):
+    """Tracks training progress and updates metrics server."""
+
+    def __init__(self, custom_metrics_map: dict, metrics_port: int = 28080):
+        self.start_time: Optional[float] = None
+        self.custom_metrics_map = custom_metrics_map
+        self.metrics_port = metrics_port
+        self.server = None
+
+    def on_train_begin(self, args, state, control, **kwargs) -> None:
```

Ruff (0.14.5)

Unused method argument: `control`

(ARG002)